### ISSUE TITLE
Option for silent or interval/triggered sync

### ISSUE THREAD

**Issue by ankrman** [strongbox-password-safe/Strongbox#495]
Hey Mark, again me (but coming from Enpass I'm constantly comparing features and behaviour between the two apps).

I turns out that the size of my DB and the speed of my DB storage is not the fastest combination in daily use, means the sync takes around 5 seconds. If I'm f.e. just changing the icon of an entry, the sync is triggered immediately and locks my interface for the beforementioned amount of time. I know I could work offline and then sync manually in the DB list but as my setup is super-multi-device-y it wouldn't be convenient either.

My idea would be to have an option either 
- to run the sync in the background without locking the GUI **or** 
- run the sync in intervals/by triggers
  - each 1, 5, 10 mins
  - app inactivity
  - losing app focus/app no longer in the foreground
  - closure of application (just to be sure)
  - whatever iOS may offer as alternative events/triggers **or**
- to toggle autosave on/off (or call it offline/online) combined with a switch button within the opened DB

I know, bringing in a switch (for autosave or off-/online) is not easy in the current interface but perhaps it's an option to introduce a bottom bar with switch button, close button, option menu (now top right), etc. Especially nowadays with all the devices getting bigger and bigger, it would be convenient to have the options reachable for normal sized hands at the bottom.

Thanks as always, keep up the great work 👍🏼

**Comment by imsdigital** [comment]
I haven’t followed the this thread so forgive me if I’m pointing out something that has already been mentioned. Occasionally I find that a version database which was saved on one device has not turned up on another device when the db is opened.

This has the potential to lose information with a lost right scenario.

Since the very nature of this information makes data loss frustrating I safeguard myself by creating a record which only contains the current date.

I’m not comfortable with it and would prefer to either be sure that the sync has completed or, preferably, have older versions save.

Just my 2c

Steve

**Comment by ankrman** [comment]
Hmm, the "open offline" function is technically not too much away from what I suggested here and theoretically has the same potential of losing information. To lose the delta information between offline and synced (online), you would explicitly have to delete the app, wouldn't you? 

Again, I could imagine that as **options**, not as the hard coded standard behaviour. As I wrote...I'm always comparing with Enpass (what my wife is using) and regarding sync, offline availability etc. there's...silence. Your just don't get any kind of information, neither when you're offline, the sync takes place, an item is saved...nothing. It just works out of the box, you're automatically offline and next time you're online the data is background synced again. 

Some could see this as intransparency or whatever, I see it as effective usage without verbose information one doesn't neccessarily need.

**Comment by strongbox-mark** [comment]
Hi @ankrman - There's some work planned which will move a lot of the encrypt/sync action into the background, and make the whole thing a bit more "Enpass" like. As to interval sync, I'm really not sure this belongs in Strongbox on iOS, but I can see it making sense on Desktop Mac. It's pretty easy to pull down to refresh/sync manually on iOS - The issue seems to be the encrypt and sync time. You could also look at reducing that encrypt time by reducing your database size (if possible?).

@imsdigital - Thanks for the report, I'm not sure this really belongs in here, it sounds like you're having some sync issues and that will be particular to your setup (Storage Provider and what other apps you use, custom sync, sync delays, other sync clients (Dropbox Desktop?) etc). It is impossible to diagnose the general issue you've mentioned without further detail. You could drop and email to support@strongboxsafe.com and we'll try take a look at your situation. You should be able to use Date Modified timestamps to see how and when changes are replicated around your devices.

This is a general issue when you manage sync yourself as opposed to having a single centralised place to store it (Bitwarden, Lastpass, 1Password).

**Comment by ankrman** [comment]
Hey @strongbox-mark, sound great. It's always better if things happen in the background like magic without having to configure everything. The interval sync stuff was only mentioned as an alternative, the "silent/background" sync is still what I'd prefer.

**Comment by strongbox-mark** [comment]
Hi @ankrman - with the release of 1.52.11, updates and syncs can be done in the background. The default for this is Off right now as the changes roll out to most users. All going well, this will be defaulted On in the next release for everyone. 

You can turn it on and test it for yourself in Preferences > Advanced > Sync Updates in Background. Feedback welcome. This should make it easier to perform multiple updates simultaneously without the UI being disabled or the spinner. I'll close this one out now, and we can address any other issues separately.

**Comment by ankrman** [comment]
Hey @strongbox-mark,

I'm already testing and loving the new behaviour.

What I've seen so far:
- I had created three entries in quick succession, then **once** ran into some network error popup (unfortunately dismissed it accidentially). But this could also have been my storage.
- Afterwards I tried to reproduce that by creating 10 entries in a row and had no issues, so it's probably nothing.

But it could be that this change somehow broke the saving/storage of icons. When I'm selecting a new icon for an entry, 
- the new icon is shown in the entry edit screen
- but is reset immediately to the default lock-icon as soon as I'm saving the entry 
- and therefore isn't changed.
